### PR TITLE
feat(main-cron,sqlsmith): increase sqlsmith fuzzing duration

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Increase queries from **100 to 1000** for sqlsmith e2e fuzzing. We have many sql functions, as well as possible inputs, so many combinations are not tested regularly if we just do 100 daily.

For this PR the e2e-runtime is ~22m. 
This happens because sqlsmith job will run for PRs changing sqlsmith source or ci file. 
In this case the modified file is `run-e2e-test.sh`, which triggers to sqlsmith job.
Normal PRs will not have this runtime.

Our daily cron job will still have the runtime increased to ~22m however, since it includes sqlsmith job.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

**Review by lmatz — APPROVED** [review]
LGTM! 

`export RUN_SQLSMITH=1; # fuzz tests` will control this 1000 queries test will only be executed in daily main-cron job, and not for every PR.

Let's see what other reviewers think

File: ci/workflows/pull-request.yml [hunk 1]
```diff
@@ -79,7 +79,7 @@ steps:
           config: ci/docker-compose.yml
           mount-buildkite-agent: true
       - ./ci/plugins/upload-failure-logs
-    timeout_in_minutes: 10
+    timeout_in_minutes: 30 # This timeout accommodates PRs with Sqlsmith job running.
```
Comment (huangjw806) [inline ci/workflows/pull-request.yml:82]: Is it necessary to do a fuzzing test in every PR? Doing so will quickly hit the buildkite job minutes limit.

Comment (kwannoel) [reply ci/workflows/pull-request.yml:82]: No, not every PR will run the fuzzing test

Comment (kwannoel) [reply ci/workflows/pull-request.yml:82]: This timeout is for PRs which have to run it, i.e. PRs which modify sqlsmith ci / src files.

Comment (huangjw806) [reply ci/workflows/pull-request.yml:82]: ok, it turns out that the fuzzing test will only be run in pr when the following files are modified.
```
MATCHES="ci/scripts/pr.env.sh\
\|ci/scripts/run-e2e-test.sh\
\|ci/scripts/run-unit-test.sh\
\|src/tests/sqlsmith"
```

**Comment by huangjw806** [comment]
> LGTM!
> 
> `export RUN_SQLSMITH=1; # fuzz tests` ensures that: this 1000 queries test will only be executed in the daily main-cron job, not for every PR.
> 
> Let's see what other reviewers think


The fuzzing test is run in pr only when some files are modified.

**Review by huangjw806 — APPROVED** [review]
LGTM

**Comment by lmatz** [comment]
Do we need another PR to change the timeout back? @kwannoel

**Comment by kwannoel** [comment]
> Do we need another PR to change the timeout back? @kwannoel

Perhaps use this approach instead: https://github.com/risingwavelabs/risingwave/pull/6059. We decouple the cron and PR fuzzing count. So even if sqlsmith runs in PR, the query count is less, and still keeps within the 10m timeout.

Otherwise each time the PR has changes to sqlsmith files, we have to:
1. Increase timeout to sqlsmith job can run to completion
2. Decrease it back to ensure normal workflows don't exceed the timeout limit.

**Pull request merged**